test: run styling benchmarks in benchpress

devversion opened the pull request:
Runs the styling benchmarks that have been added with 2e0b237
in benchpress. The goal is that these benchmarks can be wired up in
Latency Lab.

pkozlowski-opensource committed:
perf(ivy): styling algorithm benchmark

pkozlowski-opensource commented on modules/benchmarks/src/styling/ng2/index.html:
```diff
@@ -11,9 +11,10 @@
       <option value="0">(0) [title]="exp" (no styling involved)</option>
       <option value="1">(1) class="foo" (static class)</option>
-      <option value="2">(2) class="foo {{exp}}" (className binding)</option>
+      <option value="2">(2) class="foo {{exp}}" (static class with interpolation)</option>
```
nit: could we drop the "static" from the name here? It is a binding, or, to be more precise, an interpolation. How about "class interpolation"?

devversion replied:
good point. fixed.

pkozlowski-opensource commented on modules/benchmarks/src/styling/styling_perf.spec.ts:
```diff
@@ -0,0 +1,85 @@
+];
+
+/** Benchmarks that should run against all configured scenarios. */
+const BENCHMARKS = {
```
nit: do we need this const now that we always run the same work for each scenario?

devversion replied:
Not really. I kept it since I initially had more cases like `create & detect changes` and it could've been useful in the future. I removed it for now though, since I doubt there will be more soon.

pkozlowski-opensource commented on modules/benchmarks/src/styling/styling_perf.spec.ts:
```diff
@@ -0,0 +1,85 @@
+            prepare: () => {
+              // Switch to the current test mode by clicking the corresponding option.
+              element.all(by.tagName('option')).get(optionIndex).click();
+              if (init !== undefined) {
```
Are there cases where `init` is undefined?

devversion replied:
Related to my answer in the other comment. It was intended to have optional inits. e.g. when testing the `create` button, there are no preparation/init actions needed

devversion replied:
Removed for simplicity 😄

pkozlowski-opensource reviewed (approved):
LGTM overall. Left a couple of nit comments as I think that we can simplify the test a bit since we are doing the same work for all scenarios. 

Thnx @devversion !

devversion committed:
test: run styling benchmarks in benchpress

Runs the styling benchmarks that have been added with 2e0b237646863562e336f370372b4b7f9e52d818
in benchpress. The goal is that these benchmarks can be wired up in
Latency Lab.

IgorMinar commented on modules/benchmarks/src/styling/BUILD.bazel:
```diff
@@ -1 +1,15 @@
+
+ts_library(
+    name = "tests_lib",
+    testonly = 1,
```
please use `True` instead - we prefer it because of better readability.

IgorMinar replied:
I fixed this and forcepushed this this branch

devversion replied:
@IgorMinar Thanks for fixing this. I was unexpectedly out yesterday, so couldn't address it myself.

IgorMinar reviewed (approved):
the rest looks good to me. thanks!
